File: WebApplication/ClientApp/src/app/_orchestration/searchEvents/eventSearchResultDto.ts
import { LocationInfoWithRank } from "../locationInfoWithRank";
import { DisplayNameWithIdDto } from "../displayNameWithIdDto";
import { IRaceSeriesType } from "../../_common/IRaceSeriesType";

export interface EventSearchResultDto extends IRaceSeriesType {
  id: number
  name: string
  raceSeriesTypeName: string
  upcomingRaceId: number
  kickOffDate: string
  kickOffTime: string
  description: string
  courses: DisplayNameWithIdDto[]
  locationInfoWithRank: LocationInfoWithRank
  rating: number
}

File: Orchestration/SearchEvents/EventSearchResultDto.cs
namespace Orchestration.GetRaceSeriesSearch;

public record EventSearchResultDto
{
    public required List<DisplayNameWithIdDto> Courses { get; init; }
    public required string Description { get; init; }
    public required string KickOffDate { get; init; }
    public required string KickOffTime { get; init; }
    public required int Id { get; init; }
    public required LocationInfoWithRank LocationInfoWithRank { get; init; }
    public required string Name { get; init; }
    public required RaceSeriesType RaceSeriesType { get; init; }
    public required string RaceSeriesTypeName { get; init; }
    public required int Rating { get; init; }
    public required int UpcomingRaceId { get; init; }
}
